src/lib.rs

```diff
@@ -270,6 +246,62 @@ impl MemtestRunner {
 
         reports
     }
+
+    fn run_test(
+        test_kind: MemtestKind,
+        mem_lock_mode: MemLockMode,
+        memory: &mut [usize],
+        deadline: Instant,
+    ) -> Result<MemtestOutcome, MemtestError<RuntimeError>> {
+        let timeout_checker = TimeoutChecker::new(deadline);
+
+        #[cfg(unix)]
+        if matches!(mem_lock_mode, MemLockMode::PageFaultChecking) {
+            match PageFaultChecker::new(memory.as_mut_ptr() as usize, memory.len()) {
+                Ok(page_fault_checker) => test_kind.run(
+                    memory,
+                    RuntimeChecker::new(timeout_checker, page_fault_checker),
+                ),
+                Err(e) => Err(MemtestError::Other(e)),
+            }
```

Comment:
nit: Whenever I see a `match` mapping `Err` to `Err`, that smells like it should be using the [`map_err()`](https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err) function.

```Rust
let runtime_checker = RuntimeChecker::new(
    timeout_checker,
    PageFaultChecker::new(memory.as_mut_ptr() as usize, memory.len())
        .map_err(MemtestError::Other)?,
);
test_kind.run(memory, runtime_checker)
```